examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -307,7 +310,17 @@ int32_t wfx_wifi_rsi_init(void)
     if (status != SL_STATUS_OK)
     {
         SILABS_LOG("wfx_wifi_rsi_init failed %x", status);
+        return status;
     }
+
+    // Create Sempaphore for scan
+    sScanSemaphore = osSemaphoreNew(1, 0, NULL);
+    if (sScanSemaphore == NULL)
+    {
+        SILABS_LOG("Failed to create scan semaphore");
+        return SL_STATUS_FAIL;
```

Comment:
Since you're creating a dynamically allocated Semaphore, `SL_STATUS_NO_MORE_RESOURCE` or `SL_STATUS_ALLOCATION_FAILED` would be a better error code